packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -144,6 +145,40 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
                 }
         )
     }
+
+    @VisibleForTesting
+    fun onDeleteInstance(flutterResult: Result, request: HashMap<String, Any>) {
+        try {
+            var modelName = request["modelName"] as String
+            var modelData = request["model"] as HashMap<String, Any>
+            var queryOptions: QueryOptions = QueryOptionsBuilder.fromSerializedMap(request)
+            val plugin = Amplify.DataStore.getPlugin("awsDataStorePlugin") as AWSDataStorePlugin
+
+            var instance = SerializedModel.builder()
+                    .serializedData(modelData)
+                    .id(modelData["id"] as String)
+                    .modelName(modelName)
+                    .build()
+
+            plugin.delete(
+                    instance,
+                    queryOptions.queryPredicate,
+                    Consumer {
+                        LOG.info("Deleted item: " + it.item().toString())
+                        Handler(Looper.getMainLooper()).post {
```

Comment:
Can you share this a class variable between delete and query?

Reply:
done.